solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -39,9 +39,10 @@ fn get_escrow_account(
     channel_id: &ChannelId,
     denom: &str,
 ) -> Pubkey {
-    let (head, tail) = stdx::split_at::<32>(denom.as_bytes())
+    let denom = hash::hash(denom.as_bytes()).to_bytes();
+    let (head, tail) = stdx::split_at::<32>(&denom)
         .map(|(head, tail)| (&head[..], tail))
-        .unwrap_or((denom.as_bytes(), &b""[..]));
+        .unwrap_or((&denom, &b""[..]));
     let seeds = [port_id.as_bytes(), channel_id.as_bytes(), head, tail];
     let seeds = if tail.is_empty() { &seeds[..3] } else { &seeds[..] };
     Pubkey::find_program_address(seeds, &crate::ID).0
```

Comment:
No need to split since hash is always 32 bytes.
```suggestion
    let denom = CryptoHash::digest(denom.as_bytes());
    let seeds = [port_id.as_bytes(), channel_id.as_bytes(), denom.as_slice()];
    Pubkey::find_program_address(&seeds, &crate::ID).0
```